mods/tnt/init.lua

```diff
@@ -384,7 +349,74 @@ minetest.register_craft({
 	}
 })
 
+function tnt.register_tnt(def)
+	local name = ""
+	if not def.name:find(':') then
+		name = "tnt:"..def.name
+	else
+		name = def.name
+		def.name = def.name:sub(def.name:find(':')+1)
```

Comment:
I guess it matches a substring after `:` ? If yes, I would prefer `def.name:match(":([%w_]+)"`.